pkg/handler/eventhandler.go

```diff
@@ -82,10 +83,72 @@ type TypedEventHandler[object any, request comparable] interface {
 	Generic(context.Context, event.TypedGenericEvent[object], workqueue.TypedRateLimitingInterface[request])
 }
 
-var _ EventHandler = Funcs{}
+var _ EventHandler = &Funcs{}
```

Comment:
I think this can not be a pointer receiver otherwise this is a breaking change 😬 Can we also typecheck taht `TypedFuncs` which is now a different implementaiton is a `TypedEventHandler`?